sklearn/metrics/pairwise.py

```diff
@@ -1114,9 +1115,9 @@ def manhattan_distances(X, Y=None):
         Y.sum_duplicates()
         D = np.zeros((X.shape[0], Y.shape[0]))
         _sparse_manhattan(X.data, X.indices, X.indptr, Y.data, Y.indices, Y.indptr, D)
-        return D
+        return xp.asarray(D, device=device_)
 
-    return distance.cdist(X, Y, "cityblock")
```

Comment:
Note: This doesn't return the specific array and only seems to return numpy arrays.

Reply:
I don't think `scipy.spatial.distance.cdist` supports array API inputs according to:

https://scipy.github.io/devdocs/dev/api-dev/array_api.html

This means we would need to explicitly convert the input arrays to NumPy prior to passing them to `cdist`. This is what is explained here:

https://scipy.github.io/devdocs/dev/api-dev/array_api.html#adding-array-api-standard-support-to-a-scipy-function

However, `np.asarray(some_non_cpu_device_torch_tensor)` will raise:

```python
>>> import array_api_compat.torch as xp
>>> import numpy as np
>>> np.asarray(xp.ones(3, device="mps"))
Traceback (most recent call last):
  Cell In[8], line 1
    np.asarray(xp.ones(3, device="mps"))
  File ~/miniforge3/envs/dev/lib/python3.12/site-packages/torch/_tensor.py:1083 in __array__
    return self.numpy()
TypeError: can't convert mps:0 device type tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first
```

We need to use `_convert_to_numpy` for the time being (as long as [cross-device transfers via dlpack](https://github.com/data-apis/array-api/issues/626) is not adopted enough).

Reply:
This problem is actually triggered by the CUDA GPU CI in the latest run:

https://github.com/scikit-learn/scikit-learn/actions/runs/10922892103/job/30318488070?pr=29881

Reply:
Yes I saw an email which showed the results of the CUDA CI. Yeah agreed we would need to convert to numpy for now.